stl/inc/sstream

```diff
@@ -584,20 +665,25 @@ public:
         : _Mybase(_STD addressof(_Stringbuffer)), _Stringbuffer(_Str, _Mode | ios_base::in) {}
 #endif // _HAS_CXX20
 
-    basic_istringstream(basic_istringstream&& _Right) : _Mybase(_STD addressof(_Stringbuffer)) {
-        _Assign_rv(_STD move(_Right));
-    }
-
-    basic_istringstream& operator=(basic_istringstream&& _Right) noexcept /* strengthened */ {
-        _Assign_rv(_STD move(_Right));
-        return *this;
+    basic_istringstream(basic_istringstream&& _Right)
+        : _Mybase(_STD addressof(_Stringbuffer)),
+#if _HAS_CXX20
+          _Stringbuffer(_Right._Stringbuffer.get_allocator())
+#else // ^^^ _HAS_CXX20 / !_HAX_CXX20 vvv
+          _Stringbuffer(_Stringbuf_from_allocator_tag{}, _Right._Stringbuffer._Get_allocator())
+#endif // ^^^ !_HAS_CXX20 ^^^
+    {
+        _Mybase::swap(_Right);
+        _Stringbuffer._Take_contents(_STD move(_Right._Stringbuffer));

```

Comment:
Looks awkward to myself now.

Should we add internal constructors to `basic_(i,o,io)stream` which take a `basic_streambuf<_Elem, _Traits>*` and a `basic_(i,o,io)stream&&`, and then just move constructs the `_Stringbuffer`?